hail/python/hailtop/aiotools/fs/fs.py

```diff
@@ -108,7 +108,8 @@ async def open(self, url: str) -> ReadableStream:
 
     async def open_from(self, url: str, start: int, *, length: Optional[int] = None) -> ReadableStream:
         if length == 0:
-            isfile, isdir = await asyncio.gather(self.isfile(url), self.isdir(url))
+            assert not url.endswith('/')
```

Comment:
Was this not working correctly before? I'm not sure I like this change. What happens if the file ends in a slash?

Ah I see. We can't distinguish an empty file that's a directory or a file?

Reply:
This is because `isdir` fails an assert if the url doesn't end with a slash. I kind of hate that we assert that, but it seems to be a pervasive assumption. If that's the assumption, assuming we never do `open_from('gs://bucket/foo/')` also seems reasonable.